src/plugins/plugin.h

```diff
@@ -21,17 +21,27 @@ class Plugin {
 
   public:
     Plugin() = default;
+    explicit Plugin(std::string_view name) : m_name{name} {}
     virtual ~Plugin() = default;
 
-    ConvertibleProcedureAddress getFunction(std::string_view name) const {
+    [[nodiscard]] static Plugin *load(std::string_view name);
+    [[nodiscard]] static std::string createFilename(std::string_view name, bool withPrefix, bool withPostfix);
+    [[nodiscard]] static std::string createFilenameWithExtension(std::string_view name, std::string_view extension);
+
+    [[nodiscard]] ConvertibleProcedureAddress operator[](std::string_view name) const {
         return {getProcedureAddress(name)};
     }
 
+    std::string getFilename() { return m_name; }
```

Comment:
Getter that is returning attribute copy should be const

Reply:
Missed it. Done.